base/applications/cmdutils/tasklist/tasklist.c

```diff
@@ -0,0 +1,350 @@
+        if (Status != STATUS_INFO_LENGTH_MISMATCH)
+        {
+            break;
+        }
+    }
+
+    if (!NT_SUCCESS(Status))
+    {
+        // tried NT_SYSTEM_QUERY_MAX_RETRY times, or failed with some other reason
+        ConResMsgPrintf(StdOut, 0, IDS_ENUM_FAILED);
+        HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+        return FALSE;
+    }
+
+    HINSTANCE hInstance = GetModuleHandleW(NULL);
+    assert(hInstance);
+
+    ConPrintf(StdOut, L"\n");
+
+    if (!bNoHeader)
+    {
+        PrintHeader(hInstance);
+    }
+
+    PSYSTEM_PROCESS_INFORMATION pSPI;
+    pSPI = (PSYSTEM_PROCESS_INFORMATION)ProcessInfoBuffer;
+    while (pSPI)
+    {
+        PrintString(pSPI->UniqueProcessId ? pSPI->ImageName.Buffer : L"System Idle Process", COLUMNWIDTH_IMAGENAME, TRUE);
+        PrintSpace(1);
+        PrintNum((LONGLONG)(INT_PTR)pSPI->UniqueProcessId, COLUMNWIDTH_PID);
+        PrintSpace(1);
+        PrintNum((ULONGLONG)pSPI->SessionId, COLUMNWIDTH_SESSION);
+        PrintSpace(1);
+        PrintMemory(pSPI->WorkingSetSize, COLUMNWIDTH_MEMUSAGE, hInstance);
+
+        ConPrintf(StdOut, L"\n");
+
+        if (pSPI->NextEntryOffset == 0)
+            break;
+        pSPI = (PSYSTEM_PROCESS_INFORMATION)((LPBYTE)pSPI + pSPI->NextEntryOffset);
+    }
+
+    HeapFree(GetProcessHeap(), 0, ProcessInfoBuffer);
+    return TRUE;
+}
+
+INT GetArgumentType(LPCWSTR argument)
+{
+    if (argument[0] != L'/' && argument[0] != L'-')
+    {
+        return OP_PARAM_INVALID;
+    }
+    argument++;
+
+    for (INT i = 0; i < _countof(opList); i++)
+    {
+        if (!wcsicmp(opList[i], argument))
+        {
+            return i;
+        }
+    }
+    return OP_PARAM_INVALID;
+}
+
+BOOL ProcessArguments(INT argc, WCHAR **argv)
+{
+    BOOL bHasHelp = FALSE, bHasNoHeader = FALSE;
+
+    for (INT i = 1; i < argc; i++)
+    {
+        INT Argument = GetArgumentType(argv[i]);
+
+        switch (Argument)
+        {
+            case OP_PARAM_HELP:
+            {
+                if (bHasHelp)
+                {
+                    // -? already specified
+                    ConResMsgPrintf(StdOut, 0, IDS_PARAM_TOO_MUCH, argv[i], 1);
+                    ConResMsgPrintf(StdOut, 0, IDS_USAGE);
+                    return FALSE;
+                }
+                bHasHelp = TRUE;
+                break;
+            }
+            case OP_PARAM_NOHEADER:
+            {
+                if (bHasNoHeader)
+                {
+                    // -nh already specified
+                    ConResMsgPrintf(StdOut, 0, IDS_PARAM_TOO_MUCH, argv[i], 1);
+                    ConResMsgPrintf(StdOut, 0, IDS_USAGE);
+                    return FALSE;
+                }
+                bHasNoHeader = TRUE;
+                break;
+            }
+            case OP_PARAM_INVALID:
+            default:
+            {
+                ConResMsgPrintf(StdOut, 0, IDS_INVALID_OPTION);
+                ConResMsgPrintf(StdOut, 0, IDS_USAGE);
```

Comment:
```suggestion
                ConResMsgPrintf(StdErr, 0, IDS_INVALID_OPTION);
                ConResMsgPrintf(StdErr, 0, IDS_USAGE);
```

Reply:
hmmm, does ```tasklist``` on windows actually printing these messages into StdErr?

Reply:
oh yes it's true